src/ansys/edb/core/session.py

```diff
@@ -466,6 +466,12 @@ def launch_session(ansys_em_root, port_num=None):
     >>> # program goes here
     >>> session.disconnect()
     """
+    if MOD.current_session is not None:
+        if MOD.current_session.port_num == port_num:
+            return MOD.current_session
+        else:
+            raise EDBSessionException(ErrorCode.STARTUP_MULTI_SESSIONS)
+
```

Comment:
need the same change in the function below (contextmanger)

Question: what do we do if someone has a session open as a contextmanager, open a new one from here then contextmanager goes out of context? destroy the session or keep it alive?

```py

s1 = None

with session(...) as s2:
  # some ops on s2...

  # for some reason
  s1 = session(...)

### now s2 is out of scope, s1 is still in scope, but since s2 gets destroyed, so does s1.

# some ops on s1...
# will fail
```
is this the expected behavior or should it hold onto session until all references are gone?

Reply:
@hiro727 @chenchienjacklin At least for now I think it's ok to just destroy the session once we leave the scope of the with statement. To me this seems to be getting into the realm of undefined behavior/unintended usage. What do you think?

Reply:
I think the spirit of singleton and context manager is very different. Is it okay that we don't couple them together? That is, in context manager, it creates its local session and destroys it after it is out of scope. But if the user for some reason, calls launch_session within the context manager, they can still access it afterward. So that, we still keep the definition of singleton and context manager the same.

Reply:
I think we can destroy the session only if it's created in the context manager. For example,
```
s1 = launch_session()

with session(...) as s2:
  # some ops on s2...

### s2 out of scope

# some ops on s1...
# should still valid
```
I think it can be confusing if ops on `s1` are invalid after the context manager.

Reply:
remember that this change is essentially making `s1 == s2` by returning existing session. 
Won't be able to have them behave differently as is